packages/driver/src/cy/commands/navigation.ts

```diff
@@ -7,13 +7,13 @@ import $utils from '../../cypress/utils'
 import $errUtils from '../../cypress/error_utils'
 import { LogUtils } from '../../cypress/log'
 import { bothUrlsMatchAndOneHasHash } from '../navigation'
-import { $Location } from '../../cypress/location'
+import { $Location, LocationObject } from '../../cypress/location'
 
 import debugFn from 'debug'
 const debug = debugFn('cypress:driver:navigation')
 
 let id = null
-let previousDomainVisited: boolean = false
+let previousUrlVisited: LocationObject | undefined
```

Comment:
```suggestion
let previouslyVisistedUrl: LocationObject | undefined
```

Reply:
Addressed in https://github.com/cypress-io/cypress/pull/21137/commits/7500d53f24b52f1aeaa8248fa75a671e9918d7ff